diff --git a/react-native.js b/react-native.js
index 289d208c5459430856aa45abbce8e146682b0475..6bb19220698d728ca1f94b5afd2c0b6bb51f71f3 100644
--- a/react-native.js
+++ b/react-native.js
@@ -6254,9 +6254,9 @@ function onLayout(ctx, layoutParam) {
   const previousLength = state.scrollLength;
   const scrollLength = measuredLength > 0 ? measuredLength : previousLength;
   const otherAxisSize = layout[otherAxis];
-  const needsCalculate = !state.lastLayout || scrollLength > state.scrollLength || state.lastLayout.x !== layout.x || state.lastLayout.y !== layout.y;
-  state.lastLayout = layout;
   const prevOtherAxisSize = state.otherAxisSize;
+  const needsCalculate = !state.lastLayout || scrollLength !== previousLength || otherAxisSize !== prevOtherAxisSize || state.lastLayout.x !== layout.x || state.lastLayout.y !== layout.y;
+  state.lastLayout = layout;
   const didChange = scrollLength !== state.scrollLength || otherAxisSize !== prevOtherAxisSize;
   if (didChange) {
     state.scrollLength = scrollLength;
diff --git a/react-native.mjs b/react-native.mjs
index 059d2221ce5bd3a3910825cf25089e7cf5a9a9e5..2008783cb28a677b6fbb17b13c6abbc97b09dca4 100644
--- a/react-native.mjs
+++ b/react-native.mjs
@@ -6233,9 +6233,9 @@ function onLayout(ctx, layoutParam) {
   const previousLength = state.scrollLength;
   const scrollLength = measuredLength > 0 ? measuredLength : previousLength;
   const otherAxisSize = layout[otherAxis];
-  const needsCalculate = !state.lastLayout || scrollLength > state.scrollLength || state.lastLayout.x !== layout.x || state.lastLayout.y !== layout.y;
-  state.lastLayout = layout;
   const prevOtherAxisSize = state.otherAxisSize;
+  const needsCalculate = !state.lastLayout || scrollLength !== previousLength || otherAxisSize !== prevOtherAxisSize || state.lastLayout.x !== layout.x || state.lastLayout.y !== layout.y;
+  state.lastLayout = layout;
   const didChange = scrollLength !== state.scrollLength || otherAxisSize !== prevOtherAxisSize;
   if (didChange) {
     state.scrollLength = scrollLength;
diff --git a/react-native.web.js b/react-native.web.js
index b629492344227b23a0a935a088087b00f24eff76..76f7e0e2143ae27fe8d83a83476266ce2fd4806f 100644
--- a/react-native.web.js
+++ b/react-native.web.js
@@ -6918,9 +6918,9 @@ function onLayout(ctx, layoutParam) {
   const previousLength = state.scrollLength;
   const scrollLength = measuredLength > 0 ? measuredLength : previousLength;
   const otherAxisSize = layout[otherAxis];
-  const needsCalculate = !state.lastLayout || scrollLength > state.scrollLength || state.lastLayout.x !== layout.x || state.lastLayout.y !== layout.y;
-  state.lastLayout = layout;
   const prevOtherAxisSize = state.otherAxisSize;
+  const needsCalculate = !state.lastLayout || scrollLength !== previousLength || otherAxisSize !== prevOtherAxisSize || state.lastLayout.x !== layout.x || state.lastLayout.y !== layout.y;
+  state.lastLayout = layout;
   const didChange = scrollLength !== state.scrollLength || otherAxisSize !== prevOtherAxisSize;
   if (didChange) {
     state.scrollLength = scrollLength;
diff --git a/react-native.web.mjs b/react-native.web.mjs
index 22f82e42d030619814be10147f9d4ac721356611..31606fb1014066123545972ad19e0c5d19cf9091 100644
--- a/react-native.web.mjs
+++ b/react-native.web.mjs
@@ -6897,9 +6897,9 @@ function onLayout(ctx, layoutParam) {
   const previousLength = state.scrollLength;
   const scrollLength = measuredLength > 0 ? measuredLength : previousLength;
   const otherAxisSize = layout[otherAxis];
-  const needsCalculate = !state.lastLayout || scrollLength > state.scrollLength || state.lastLayout.x !== layout.x || state.lastLayout.y !== layout.y;
-  state.lastLayout = layout;
   const prevOtherAxisSize = state.otherAxisSize;
+  const needsCalculate = !state.lastLayout || scrollLength !== previousLength || otherAxisSize !== prevOtherAxisSize || state.lastLayout.x !== layout.x || state.lastLayout.y !== layout.y;
+  state.lastLayout = layout;
   const didChange = scrollLength !== state.scrollLength || otherAxisSize !== prevOtherAxisSize;
   if (didChange) {
     state.scrollLength = scrollLength;
